virttest/utils_misc.py

```diff
@@ -851,6 +851,43 @@ def archive_as_tarball(source_dir, dest_dir, tarball_name=None,
     tarball.close()
 
 
+def get_guest_service_status(session, service, service_former=None):
+    """
+    Get service's status in guest.
+    Return 'active' for 'running' and 'active' case.
+    Return 'inactive' for 'stopped' and 'inactive' case.
+
+    :param session: An Expect or ShellSession instance to operate on
+    :type devices: Expect or ShellSession class
+    :param service: Service name that we want to check status.
+    :type service: String
+    :param service_former: Service's former name
+    :type service: String
+    :return: 'active' or 'inactive'
+    :rtype: String
+    """
+    output = ""
+    cmd = ("service %(key)s status || "
+           "systemctl %(key)s.service status || "
+           "status %(key)s")
+    try:
+        output = session.cmd_output(cmd % {'key': service})
+    except Exception:
+        if not service_former:
+            raise exceptions.TestError("Fail to get %s status.\n%s" %
+                                       (service, output))
+        output = session.cmd_output(cmd % {'key': service_former})
+
+    if re.search("running|active", output, re.I):
```

Comment:
It looks we need to search multi-lines, output of systemctl status httpd on my host likes below:
[xutian@dhcp-10-17]$ systemctl status httpd 
● httpd.service - The Apache HTTP Server
   Loaded: loaded (/usr/lib/systemd/system/httpd.service; disabled; vendor preset: disabled)
   Active: active (running) since Tue 2015-12-15 13:32:57 CST; 1 day 21h ago
  Process: 15807 ExecReload=/usr/sbin/httpd $OPTIONS -k graceful (code=exited, status=0/SUCCESS)
 Main PID: 9002 (httpd)
   Status: "Total requests: 6; Idle/Busy workers 100/0;Requests/sec: 3.68e-05; Bytes served/sec: 9.0KB/sec"
   CGroup: /system.slice/httpd.service
           ├─ 9002 /usr/sbin/httpd -DFOREGROUND
           ├─15809 /usr/sbin/httpd -DFOREGROUND
           ├─15810 /usr/sbin/httpd -DFOREGROUND
           ├─15811 /usr/sbin/httpd -DFOREGROUND
           ├─15813 /usr/sbin/httpd -DFOREGROUND
           └─15815 /usr/sbin/httpd -DFOREGROUND

Dec 15 13:32:57 dhcp-10-17.nay.redhat.com systemd[1]: Starting The Apache HTTP Server...
Dec 15 13:32:57 dhcp-10-17.nay.redhat.com systemd[1]: Started The Apache HTTP Server.
Dec 16 03:08:02 dhcp-10-17.nay.redhat.com systemd[1]: Reloaded The Apache HTTP Serve